server/src/main/kotlin/suwayomi/tachidesk/global/impl/util/Jwt.kt

```diff
@@ -0,0 +1,97 @@
+package suwayomi.tachidesk.global.impl.util
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.JWTVerifier
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.JWTVerificationException
+import io.github.oshai.kotlinlogging.KotlinLogging
+import suwayomi.tachidesk.server.user.UserType
+import java.security.SecureRandom
+import java.time.Instant
+import javax.crypto.spec.SecretKeySpec
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+import kotlin.time.Duration.Companion.days
+import kotlin.time.Duration.Companion.hours
+
+object Jwt {
+    private val logger = KotlinLogging.logger {}
+
+    private const val ALGORITHM = "HmacSHA256"
+    private val accessTokenExpiry = 1.hours
+    private val refreshTokenExpiry = 60.days
+    private const val ISSUER = "tachidesk"
+    private const val AUDIENCE = "" // todo audience
```

Comment:
This TODO should probably be investigated(I had not gotten to it yet)

Reply:
From a quick search, this should usually be the public url of the API, but this is not really known here. We could make it a configurable setting? Another option is also just some generic string like "tenant-xyz" or "service-A", so the setting could really be any string, maybe defaulting to "suwayomi-server".

https://mojoauth.com/blog/lets-understand-jwt-audience-aud-claim/

Reply:
That works for me